akka-bench-jmh/src/main/scala/akka/stream/MaterializationBenchmark.scala

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.stream
+
+import java.util.concurrent.TimeUnit
+import akka.actor.ActorSystem
+import akka.stream.scaladsl._
+import org.openjdk.jmh.annotations._
+
+object MaterializationBenchmark {
+
+  val complexGraphBuilder = () =>
+    FlowGraph.closed() { implicit b =>
+      val numOfJunctions = 100
+
+      import FlowGraph.Implicits._
+      val broadcast = b.add(Broadcast[Unit](numOfJunctions))
+      var outlet = broadcast.out(0)
+      for (i <- 1 until numOfJunctions) {
+        val merge = b.add(Merge[Unit](2))
+        outlet ~> merge
+        broadcast.out(i) ~> merge
+        outlet = merge.out
+      }
+
+      Source.single(()) ~> broadcast
+      outlet ~> Sink.ignore
+    }
+
+  val graphWithNestedImportsBuilder = () => {
+    var flow: Graph[FlowShape[Unit, Unit], Unit] = Flow[Unit].map(identity)
+    for (_ <- 1 to 100) {
+      flow = FlowGraph.partial(flow) { implicit b ⇒
+        flow ⇒
+          FlowShape(flow.inlet, flow.outlet)
+      }
+    }
+
+    FlowGraph.closed(flow) { implicit b ⇒
+      flow ⇒
+        import FlowGraph.Implicits._
+        Source.single(()) ~> flow ~> Sink.ignore
+    }
+  }
+}
+
+@State(Scope.Benchmark)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@BenchmarkMode(Array(Mode.Throughput))
+class MaterializationBenchmark {
+  import MaterializationBenchmark._
+
+  implicit val system = ActorSystem("MaterializationBenchmark")
+  implicit val mat = ActorFlowMaterializer()
+
+  val simpleFlow = Source.single(()).map(identity).to(Sink.ignore)
+
+  val complexFlow = {
+    var source = Source.single(())
+    for (_ <- 1 to 100) {
+      source = source.map(identity)
```

Comment:
The loop wires 100 maps in sequence. And this code is not part of the benchmark.